store/localcachelayer/layer.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package localcachelayer
+
+import (
+	"github.com/mattermost/mattermost-server/einterfaces"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mattermost-server/store"
+	"github.com/mattermost/mattermost-server/utils"
+)
+
+const (
+	REACTION_CACHE_SIZE = 20000
+	REACTION_CACHE_SEC  = 30 * 60
+
+	CLEAR_CACHE_MESSAGE_DATA = ""
+)
+
+type LocalCacheStore struct {
+	store.Store
+	metrics       einterfaces.MetricsInterface
+	cluster       einterfaces.ClusterInterface
+	reaction      LocalCacheReactionStore
+	reactionCache *utils.Cache
+}
+
+func NewLocalCacheLayer(baseStore store.Store, metrics einterfaces.MetricsInterface, cluster einterfaces.ClusterInterface) LocalCacheStore {
+	localCacheStore := LocalCacheStore{
+		Store:   baseStore,
+		cluster: cluster,
+		metrics: metrics,
+	}
+	localCacheStore.reactionCache = utils.NewLruWithParams(REACTION_CACHE_SIZE, "Reaction", REACTION_CACHE_SEC, model.CLUSTER_EVENT_INVALIDATE_CACHE_FOR_REACTIONS)
+	localCacheStore.reaction = LocalCacheReactionStore{ReactionStore: baseStore.Reaction(), rootStore: &localCacheStore}
+
+	if cluster != nil {
+		cluster.RegisterClusterMessageHandler(model.CLUSTER_EVENT_INVALIDATE_CACHE_FOR_REACTIONS, localCacheStore.reaction.handleClusterInvalidateReaction)
```

Comment:
nit: spurious extra newline

Reply:
where? I use that line to separate two blocks, the caches initialization and the cluster event message handlers registration.

Reply:
Hmm, not seeing this one anymore either. Please disregard :)